neo4j/connector.go

```diff
@@ -139,6 +140,9 @@ func (c *connector) wrapInTls(target string, rawConn net.Conn) (net.Conn, error)
 	tlsConn := tls.Client(rawConn, &conf)
 	err = tlsConn.Handshake()
 	if err != nil {
+		if err == io.EOF {
+			err = errors.New("Remote end closed the connection, check that TLS is enabled on the server")
```

Comment:
Either it is that TLS should be enabled on the server or that the client config should be set to not to use an encrypted connection. But yes it is nice to get an error message that hints that it is probably something to do with TLS settings.

Reply:
Yes, I assume the client is right ;-)